Makefile

```diff
@@ -113,6 +113,59 @@ manifests-gen-crds: controller-gen yq
 	for YAMLFILE in config/crd/bases/actions*.yaml; do \
 		$(YQ) '.spec.preserveUnknownFields = false' --inplace "$$YAMLFILE" ; \
 	done
+	#runners
```

Comment:
We encountered a bunch of `Invalid value: "object": must be a scalar or atomic type as item of a list with x-kubernetes-list-type=set` errors on CRD deployment due to the controller-gen is somehow generating an invalid CRD manifests, where the `x-kubernetes-list-type` is `set` while the `items.type` is `object` and Kubernetes API server rejecting those.

The items actually look like objects, so probably we need any of (1)remove `x-kubernetes-list-type=set` from the generated CRD manifests (2)fix controller-gen to not set `x-kubbernetes-list-type=set` for some embedded fields (3)fix Kubernetes to support `x-kubernetes-list-type=set` with `object` items.

I opted to go with (1), as it looked like the fastest and the easiest solution. ZThis way, we don't need to fork Kubernetes or controller-tools just to address the issue. Also, we can easily remove these yq commands once the issue is resolved in upstream.

Reply:
See https://github.com/VictoriaMetrics/operator/issues/37 for a similar issue.
I couldn't find any upstream issues related to this.

Reply:
Trying to see whether we make some function to run through the folder and fix all the files, so we don't have to hardcode 30+ lines of file here.

Reply:
how do you figure out which one needs to be deleted?

Reply:
@TingluoHuang I figured it out by running go-tests until I finally saw no errors. The errors occurred on applying CRD manifests to the test kube-apiserver in any controller-runtime's envtest-managed tests.

Every error seems to be coming from the `resources` field definition under embedded `corev1.Container` and `corev1.Volume` fields like `containers`, `initContainers`, `sidecarContainers`, `volumes`, `volumeClaimTemplate`, `workVolumeClaimTemplate`, and so on.

That said, it might not be straightforward to write up a simple script to replace those hardcoded 30+ lines.
I wish there's a better way though!